src/GovUk.Education.ExploreEducationStatistics.Common/Services/BlobStorageService.cs

```diff
@@ -350,43 +350,47 @@ private static async Task<string> UploadToTemporaryFile(IFormFile file)
             return targetStream;
         }
 
-        public async Task<Stream> StreamBlob(IBlobContainer containerName, string path, int? bufferSize = null)
+        public async Task<Stream> StreamBlob(
+            IBlobContainer containerName,
+            string path,
+            int? bufferSize = null,
+            CancellationToken cancellationToken = default)
         {
-            // Azure SDK v12 isn't compatible with how we want to use file
-            // streams i.e. they need to be seekable. This is particularly
-            // a problem for MIME type validation.
-            // See: https://github.com/Azure/azure-sdk-for-net/pull/15032
-            // TODO: Change to v12 implementation when possible
-            var blobContainer = await GetCloudBlobContainer(containerName);
-            var blob = blobContainer.GetBlockBlobReference(path);
+            var blob = await GetBlobClient(containerName, path);
 
-            if (!await blob.ExistsAsync())
+            try
             {
-                throw new FileNotFoundException($"Could not find file at {containerName}/{path}");
+                return await blob.OpenReadAsync(bufferSize: bufferSize, cancellationToken: cancellationToken);
             }
-
-            if (bufferSize != null)
+            catch (RequestFailedException exception)
             {
-                blob.StreamMinimumReadSizeInBytes = (int) bufferSize;
-            }
+                if (exception.Status == 404)
+                {
+                    ThrowFileNotFoundException(containerName, path);
+                }
 
-            return await blob.OpenReadAsync();
+                throw;
+            }
         }
 
         public async Task<string> DownloadBlobText(IBlobContainer containerName, string path)
         {
             var blob = await GetBlobClient(containerName, path);
 
-            if (!await blob.ExistsAsync())
+            try
             {
-                throw new FileNotFoundException($"Could not find file at {containerName}/{path}");
+                var response = await blob.DownloadContentAsync();
+                return response.Value.Content.ToString();
```

Comment:
Cool, I'm just wondering what `Value` is here because this differs slightly from the docs linked to in the description?

Reply:
The difference with the docs is that the docs implicitly convert the `Response<BlobDownloadResult>` to a `DownloadResult` by specifying the type.

Without specifying the type, we need to call `Response.Value` to unwrap the `BlobDownloadResult`.

Reply:
Ah yeah, thanks for the explanation about the implicit conversion. 🙂